llvm/lib/Transforms/Utils/Evaluator.cpp

```diff
@@ -124,6 +125,20 @@ isSimpleEnoughValueToCommit(Constant *C,
   return isSimpleEnoughValueToCommitHelper(C, SimpleConstants, DL);
 }
 
+static std::pair<bool, GlobalValue::LinkageTypes>
+getPruningAndLinkage(const GlobalVariable &GV) {
```

Comment:
This function does two jobs which I think would be useful to separate: 
- It returns  whether a GV was pruned.
- It returns the GV's original linkage?

The caller doesn't depend on this all being implementation in a single function, I'd be tempted to have a function `bool wasPruned(const GlobalVariable&)` and another `GlobalValue::LinkageType getOriginalLinkage (const GlobalVariable&, bool WasPruned)`. This also has the benefit that we can initialize the `Pruned` and `InitialLinkage` variables in the caller at the point of defining the variable.

Reply:
> 
> 
>     * It returns the GV's original linkage?
Yes.

Good point. I have updated the patch following your advice.